plugin/builtin/helmchartinflationgenerator/HelmChartInflationGenerator_test.go

```diff
@@ -579,3 +581,201 @@ valuesInline:
+				chartDir = fmt.Sprintf("charts/%s-%s/%s", tt.chartName, tt.version, tt.chartName)
+			} else {
+				chartDir = fmt.Sprintf("charts/%s", tt.chartName)
+			}
+
+			fmt.Printf("%s: %s\n", tt.name, chartDir)
+
+			d, err := th.GetFSys().ReadFile(filepath.Join(th.GetRoot(), chartDir, "Chart.yaml"))
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			assert.Contains(t, string(d), fmt.Sprintf("name: %s", tt.chartName))
+			if tt.version != "" {
+				assert.Contains(t, string(d), fmt.Sprintf("version: %s", tt.version))
+			}
+		})
+	}
+}
+
+// Test that verifies +1 instances of same chart with different versions
+// https://github.com/kubernetes-sigs/kustomize/issues/4813
+func TestHelmChartInflationGeneratorWithMultipleInstancesSameChartDifferentVersions(t *testing.T) {
+	th := kusttest_test.MakeEnhancedHarnessWithTmpRoot(t).
+		PrepBuiltin("HelmChartInflationGenerator")
+	defer th.Reset()
+	if err := th.ErrIfNoHelm(); err != nil {
+		t.Skip("skipping: " + err.Error())
+	}
+
+	podinfo1 := th.LoadAndRunGenerator(`
+apiVersion: builtin
+kind: HelmChartInflationGenerator
+metadata:
+ name: podinfo
+name: podinfo
+version: 6.2.1
+repo: https://stefanprodan.github.io/podinfo
+releaseName: podinfo1
+`)
+
+	podinfo2 := th.LoadAndRunGenerator(`
+apiVersion: builtin
+kind: HelmChartInflationGenerator
+metadata:
+ name: podinfo
+name: podinfo
+version: 6.1.8
+repo: https://stefanprodan.github.io/podinfo
+releaseName: podinfo2
+`)
+
+	podinfo1Img, err := podinfo1.Resources()[1].GetFieldValue("spec.template.spec.containers.0.image")
+	assert.NoError(t, err)
+	assert.Equal(t, "ghcr.io/stefanprodan/podinfo:6.2.1", podinfo1Img)
+
+	podinfo2Img, err := podinfo2.Resources()[1].GetFieldValue("spec.template.spec.containers.0.image")
+	assert.NoError(t, err)
+	assert.Equal(t, "ghcr.io/stefanprodan/podinfo:6.1.8", podinfo2Img)
+
+	podinfo1ChartsDir := filepath.Join(th.GetRoot(), "charts/podinfo-6.2.1/podinfo")
+	assert.True(t, th.GetFSys().Exists(podinfo1ChartsDir))
+
+	podinfo2ChartsDir := filepath.Join(th.GetRoot(), "charts/podinfo-6.1.8/podinfo")
+	assert.True(t, th.GetFSys().Exists(podinfo2ChartsDir))
+
+	podinfo1ChartContents, err := th.GetFSys().ReadFile(filepath.Join(podinfo1ChartsDir, "Chart.yaml"))
+	assert.NoError(t, err)
+	assert.Contains(t, string(podinfo1ChartContents), "version: 6.2.1")
+
+	podinfo2ChartContents, err := th.GetFSys().ReadFile(filepath.Join(podinfo2ChartsDir, "Chart.yaml"))
+	assert.NoError(t, err)
+	assert.Contains(t, string(podinfo2ChartContents), "version: 6.1.8")
+
+}
+
+// Addressed: https://github.com/kubernetes-sigs/kustomize/issues/5163
+func TestHelmChartInflationGeneratorWithLocalChartWithVersion(t *testing.T) {
+	th := kusttest_test.MakeEnhancedHarnessWithTmpRoot(t).
+		PrepBuiltin("HelmChartInflationGenerator")
+	defer th.Reset()
+	if err := th.ErrIfNoHelm(); err != nil {
+		t.Skip("skipping: " + err.Error())
+	}
+
+	err := th.GetFSys().MkdirAll(filepath.Join(th.GetRoot(), "charts/dummy/templates"))
+	require.NoError(t, err)
```

Comment:
can we use assert here? so that we don't need to import it. for consistency as well

Reply:
sure, updated!